resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1162,7 +1165,11 @@ private[spark] class Client(
       val state = report.getYarnApplicationState
 
       if (logApplicationReport) {
-        logInfo(s"Application report for $appId (state: $state)")
+        if (reportsSinceLastLog == reportsTillNextLog || lastState != state) {
```

Comment:
Right now this can't happen, but I'm usually paranoid about using equality checks in case there is some bug where we don't hit this line of code, but do increment `reportsSinceLastLog`. A more defensive way to code this is `reportsSinceLastLog >= reportsTillNextLog`.